src/deprecation.cpp

```diff
@@ -12,6 +12,11 @@
 static const std::string CLIENT_VERSION_STR = FormatVersion(CLIENT_VERSION);
 
 void EnforceNodeDeprecation(int nHeight, bool forceLogging) {
+
+    // Do not enforce deprecation in regtest or on testnet
+    bool notMainNet = (GetBoolArg("-testnet", false) || GetBoolArg("-regtest", false));
```

Comment:
This could technically be invalid if the argument-parsing code had a bug, whereas checking the network string is more reliable. In reality, it probably doesn't make a difference, so I'll wait and see what the next reviewer thinks.